src/main/java/com/facebook/openwifirrm/modules/ModelerUtils.java

```diff
@@ -381,4 +399,269 @@ public static Map<String, Map<String, WifiScanEntry>> getAggregatedWifiScans(
 		}
 		return aggregatedWifiScans;
 	}
+
+	/** This method converts the agggregated map to a Json String.
+	 *
+	 * @param aggregatedMap 	Aggregated map
+	 * @param aggType 			Determines how to aggregate the statistics.
+	 * @return String			A Json String representation of the aggregated map.
+	 */
+	public static String aggregatedStatesToJson(
+		Map<String, Map<String, Map<String, Map<JsonNode, Map<String, MeanAggregator>>>>> aggregatedMap,
+		AggregationType aggType
+	) {
+		GsonBuilder gsonBuilder = new GsonBuilder();
+
+		Type AggregationType = new TypeToken<MeanAggregator>() {}.getType();
+		JsonSerializer<MeanAggregator> serializer;
+		/**
+		 * Custom the json serializer by overriding the serialize method.
+		*/
+		switch (aggType) {
+		case LIST:
+			serializer =
+				new JsonSerializer<MeanAggregator>() {
+					/**
+					 * If the aggregation type is to take list of the values,
+					 * return a JsonArray of the value list.
+					 */
+					@Override
+					public JsonElement serialize(
+						MeanAggregator agg,
+						java.lang.reflect.Type typeOfSrc,
+						JsonSerializationContext context
+					) {
+						JsonArray aggJson = new JsonArray();
+						for (Double value : agg.getList()) {
```

Comment:
You can use `Gson` to convert an `ArrayList` to `JsonArray`.

Reply:
I guess it might be the same if I initial it as a JsonArray or ArrayList?

Reply:
I think you are converting an `ArraryList` or `LinkedList` to a `JsonArray` here, aren't you? I believe `Gson` have some methods to deal with it directly.